proto/proto-backplane-grpc/src/main/proto/deephaven/proto/table.proto

```diff
@@ -748,6 +753,20 @@ message AjRajTablesRequest {
   repeated string columns_to_add = 6;
 }
 
+message MultiJoinInput {
+  TableReference source_id = 1;
+  repeated string columns_to_match = 3;
+  repeated string columns_to_add = 4;
+}
+
+message MultiJoinTablesRequest {
+  Ticket result_id = 1;
+  // It is considered an error if both `source_ids` and `multi_join_inputs` are provided.
```

Comment:
As a comment, this will appear on the `source_ids` field in each language, rather than be general for the body of the message. Consider moving to just before the message itself, or duplicate it to the `multi_join_inputs` as well?

Reply:
...that said, this might be better expressed as a `oneof`? Reading the logic in the java server impl, if `multi_join_inputs` is provided, neither `source_ids` nor `columns_to_match` can be specified (i.e. "will be empty")?

I'm not above just making the runtime logic encode this instead of the message format, but there is something to be said for making the serialization api simply _not support_ invalid states.

Reply:
as @devinrsmith pointed out, it's really `(source_ids + columns_to_match) ^ multi_join_inputs` that needs to be enforced.  `oneof` doesn't seem to work here without adding another nesting layer which I think would break backward compatibility.

Reply:
What do you mean "break backward compatibility" - this is ostensibly a brand new message, yeah?

Reply:
Good point.  RangeJoin is published, MultiJoin is new.